File: docs/visualizations/examples/screenshot.js
const { chromium } = require('playwright');

(async () => {
    const browser = await chromium.launch();
    const pages = [
        { file: 'sample-dependency-map.html', out: 'screenshot-dependency-map.png', waitMs: 2000 },
        { file: 'sample-complexity-dashboard.html', out: 'screenshot-complexity-dashboard.png', waitMs: 500 },
        { file: 'sample-comparative-table.html', out: 'screenshot-comparative-table.png', waitMs: 500 },
    ];

    for (const p of pages) {
        const page = await browser.newPage({ viewport: { width: 1000, height: 800 } });
        await page.goto(`file://${__dirname}/${p.file}`, { waitUntil: 'networkidle' });
        // Wait for D3 force simulation to settle
        await page.waitForTimeout(p.waitMs);
        // Screenshot just the article content
        const article = await page.$('article');
        if (article) {
            await article.screenshot({ path: `${__dirname}/${p.out}` });
        } else {
            await page.screenshot({ path: `${__dirname}/${p.out}`, fullPage: true });
        }
        await page.close();
        console.log(`Captured: ${p.out}`);
    }

    await browser.close();
})();

File: docs/visualizations/examples/screenshot.py
#!/usr/bin/env python3
"""Take screenshots of sample HTML visualizations using Chrome headless."""
import os
import subprocess
import time

DIR = os.path.dirname(os.path.abspath(__file__))
CHROME = '/root/.cache/ms-playwright/chromium-1194/chrome-linux/chrome'

pages = [
    ('sample-dependency-map.html', 'screenshot-dependency-map.png', 1000, 750),
    ('sample-complexity-dashboard.html', 'screenshot-complexity-dashboard.png', 1000, 600),
    ('sample-comparative-table.html', 'screenshot-comparative-table.png', 900, 700),
]

for html_file, out_file, w, h in pages:
    url = f'file://{os.path.join(DIR, html_file)}'
    out_path = os.path.join(DIR, out_file)
    cmd = [
        CHROME,
        '--headless=new',
        '--no-sandbox',
        '--disable-gpu',
        '--disable-software-rasterizer',
        f'--window-size={w},{h}',
        f'--screenshot={out_path}',
        '--hide-scrollbars',
        '--virtual-time-budget=3000',
        url,
    ]
    result = subprocess.run(cmd, capture_output=True, text=True, timeout=15)
    if os.path.exists(out_path):
        print(f'Captured: {out_file} ({os.path.getsize(out_path)} bytes)')
    else:
        print(f'Failed: {out_file}')
        print(result.stderr[-500:] if result.stderr else 'no stderr')
